x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -2003,6 +2006,30 @@ private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
         future.actionGet();
     }
 
+    public Map<String, String> getAuthContextForSlowLog() {
```

Comment:
This is "leaking" internals into the slow log. I decided against doing any mapping here (map `user.realm` value `_service_account` to `SERVICE_ACCOUNT` for example) to make it easier to maintain. The benefits of doing some sort of mapping is that the log could be slightly easier to read.

Reply:
I think that is fine ... the values in the log don't need to be pretty nor adhere to any longer term contract.